include/solana.hpp

```diff
@@ -94,56 +70,40 @@ struct Instruction {
   std::vector<uint8_t> data;
 };
 
-struct CompactU16 {
-  static void encode(uint16_t num, std::vector<uint8_t> &buffer) {
-    buffer.push_back(num & 0x7f);
-    num >>= 7;
-    if (num == 0) return;
-
-    buffer.back() |= 0x80;
-    buffer.push_back(num & 0x7f);
-    num >>= 7;
-    if (num == 0) return;
-
-    buffer.back() |= 0x80;
-    buffer.push_back(num & 0x3);
-  };
-
-  static void encode(const std::vector<uint8_t> &vec,
-                     std::vector<uint8_t> &buffer) {
-    encode(vec.size(), buffer);
-    buffer.insert(buffer.end(), vec.begin(), vec.end());
-  }
-};
+namespace CompactU16 {
+void encode(uint16_t num, std::vector<uint8_t> &buffer);
+
+void encode(const std::vector<uint8_t> &vec, std::vector<uint8_t> &buffer);
+};  // namespace CompactU16
+
 struct Blockhash {
   PublicKey publicKey;
   uint64_t lastValidBlockHeight;
 };
+
+struct Balance {
+  long lamports;
+};
+
+struct SimulateTransactionResponse {
```

Comment:
all API response types should correspond to web3.js:

getBalance -> number (u64)
simulateTransaction -> Simulate**d**TransactionResponse